Cleanup fragile 'capitalize layer names' option

nyalldawson opened the pull request:
This option was being applied in the wrong place - within the map layer classes themselves. This meant that depending on the user's setting for this option, a plugin calling QgsMapLayer::setName would not be guaranteed the same behaviour across installs. (and the same with setDataSource)

Similarly, the option was re-applied on project load, so moving projects between installs with different values for this setting would affect the project layer names, breaking expressions which relied on these...

In other words, exactly the kind of user setting we want to avoid!

Instead, move the formatting and capitalization of layer names to the QgisApp add*Layer methods instead, so this option only applies on adding new layers to a project.

I've followed this up with some other related tweaks:
- Remove the capitalize setting from options and make it always on instead (it's nice to have a better auto-generated layer name, since it saves user work!)
- Show vector layer WKB types in the layer tree tooltip, and remove this from auto-generated layer names

nyalldawson committed:
Cleanup fragile 'capitalize layer names' option

This option was being applied in the wrong place - within the
map layer classes themselves. This meant that depending on the
user's setting for this option, a plugin calling QgsMapLayer::setName
would not be guaranteed the same behaviour across installs.
(and the same with setDataSource)

Similarly, the option was re-applied on project load, so
moving projects between installs with different values for
this setting would affect the project layer names, breaking
expressions which relied on these...

Instead, move the formatting and capitalization of layer
names to the QgisApp add*Layer methods instead, so this option
only applies on adding new layers to a project.

nyalldawson committed:
Remove QgsMapLayer::originalName()

Now QgsMapLayer::name() is guaranteed to return the same string

nyalldawson committed:
Show vector layer wkb types in tooltip in layer tree and layer combo boxes

This is very useful information to have readily visible (esp.
to see if a layer is multi* type, has z/m, when choosing it
as a source for a processing algorithm)

Remove redundant geometry type display from default layer name

nyalldawson committed:
Tweaks to automatic layer name logic

NathanW2 commented:
+1

On Sun., 12 Nov. 2017, 4:41 pm Nyall Dawson, <notifications@github.com>
wrote:

> This option was being applied in the wrong place - within the map layer
> classes themselves. This meant that depending on the user's setting for
> this option, a plugin calling QgsMapLayer::setName would not be guaranteed
> the same behaviour across installs. (and the same with setDataSource)
>
> Similarly, the option was re-applied on project load, so moving projects
> between installs with different values for this setting would affect the
> project layer names, breaking expressions which relied on these...
>
> In other words, exactly the kind of user setting we want to avoid!
>
> Instead, move the formatting and capitalization of layer names to the
> QgisApp add*Layer methods instead, so this option only applies on adding
> new layers to a project.
>
> I've followed this up with some other related tweaks:
>
>    - Remove the capitalize setting from options and make it always on
>    instead (it's nice to have a better auto-generated layer name, since it
>    saves user work!)
>    - Show vector layer WKB types in the layer tree tooltip, and remove
>    this from auto-generated layer names
>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/qgis/QGIS/pull/5608
> Commit Summary
>
>    - Cleanup fragile 'capitalize layer names' option
>    - Remove QgsMapLayer::originalName()
>    - Remove option to capitalize layer names, and make it always on
>    - Show vector layer wkb types in tooltip in layer tree and layer combo
>    boxes
>    - Tweaks to automatic layer name logic
>
> File Changes
>
>    - *M* doc/api_break.dox
>    <https://github.com/qgis/QGIS/pull/5608/files#diff-0> (2)
>    - *M* python/core/qgsmaplayer.sip
>    <https://github.com/qgis/QGIS/pull/5608/files#diff-1> (14)
>    - *M* src/app/qgisapp.cpp
>    <https://github.com/qgis/QGIS/pull/5608/files#diff-2> (33)
>    - *M* src/app/qgsoptions.cpp
>    <https://github.com/qgis/QGIS/pull/5608/files#diff-3> (7)
>    - *M* src/app/qgsrasterlayerproperties.cpp
>    <https://github.com/qgis/QGIS/pull/5608/files#diff-4> (4)
>    - *M* src/app/qgsvectorlayerproperties.cpp
>    <https://github.com/qgis/QGIS/pull/5608/files#diff-5> (4)
>    - *M* src/core/layertree/qgslayertreemodel.cpp
>    <https://github.com/qgis/QGIS/pull/5608/files#diff-6> (7)
>    - *M* src/core/qgsmaplayer.cpp
>    <https://github.com/qgis/QGIS/pull/5608/files#diff-7> (31)
>    - *M* src/core/qgsmaplayer.h
>    <https://github.com/qgis/QGIS/pull/5608/files#diff-8> (19)
>    - *M* src/core/qgsmaplayermodel.cpp
>    <https://github.com/qgis/QGIS/pull/5608/files#diff-9> (8)
>    - *M* src/core/qgsvectorlayer.cpp
>    <https://github.com/qgis/QGIS/pull/5608/files#diff-10> (7)
>    - *M* src/core/raster/qgsrasterlayer.cpp
>    <https://github.com/qgis/QGIS/pull/5608/files#diff-11> (4)
>    - *M* src/ui/qgsoptionsbase.ui
>    <https://github.com/qgis/QGIS/pull/5608/files#diff-12> (185)
>    - *M* tests/src/core/testqgsmaplayer.cpp
>    <https://github.com/qgis/QGIS/pull/5608/files#diff-13> (9)
>
> Patch Links:
>
>    - https://github.com/qgis/QGIS/pull/5608.patch
>    - https://github.com/qgis/QGIS/pull/5608.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/qgis/QGIS/pull/5608>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAXS3Ow9D_8UvKvqokK7xf3LVs-tacvSks5s1pMdgaJpZM4QaxyQ>
> .
>

nyalldawson committed:
[needs-docs] Remove option to capitalize layer names, and make it always on

Doing this removes some work from our users - giving layers
a nicely generated name by default

semiautomaticgit commented:
Hello and thank you for your great work.
I would propose to set this function as optional (please read https://issues.qgis.org/issues/17741).
This function could be confusing to the user when working with satellite images that include underscore and lower case letters. Also, it overrides the name when loading a layer through Python APIs.
Thank you.

Best,
Luca

nirvn commented:
I have also not been able to fully adapt / enjoy this change. Forced capitalization feels very intrusive at times.
